apps/web/app/api/[...route]/route.ts

```diff
@@ -95,16 +96,87 @@ const createR2Client = () => {
   });
 };
 
-const api = new Hono()
-  .get(
-    "/accounts/:user_id/checkout",
-    zValidator(
-      "query",
-      z.object({
-        plan: PricingPlanId,
-        interval: PricingPlanInterval,
-      })
-    ),
+const commonErrorResponses = {
+  400: {
+    description: "TODO",
+    content: {
+      "application/json": {
+        schema: z.object({
+          error: z.string().openapi({
+            description: "The error that occured",
+          }),
+        }),
+      },
+    },
+  },
+  401: {
+    description: "TODO",
+    content: {
+      "application/json": {
+        schema: z.object({
+          error: z.string().openapi({
+            description: "The error that occured",
+          }),
+        }),
+      },
+    },
+  },
+  500: {
+    description: "Internal Server Errorr",
+    content: {
+      "application/json": {
+        schema: z.object({
+          error: z.string().openapi({
+            description: "The error that occured",
+          }),
+        }),
+      },
+    },
+  },
+} as const;
+
+const api = new OpenAPIHono()
+  .doc("/openapi.json", {
+    openapi: "3.0.0",
+    info: {
+      title: "Zenblog API",
+      version: "1.0.0",
+    },
+    servers: [{ url: "https://api.zenblog.com", description: "Production" }],
+  })
+  .openapi(
+    createRoute({
+      method: "get",
+      path: "/accounts/:user_id/checkout",
+      request: {
+        params: z.object({
+          user_id: z.string().openapi({ description: "TODO" })
+        }),
+        query: z.object({
+          plan: PricingPlanId.openapi({
+            description: "TODO",
+          }),
+          interval: PricingPlanInterval.openapi({
+            description: "TODO",
+          }),
+        }),
+      },
+      responses: {
+        200: {
+          description: "TODO",
+          content: {
+            "application/json": {
+              schema: z.object({
+                url: z.string().openapi({
+                  description: "A stripe checkout url",
+                }),
+              }),
+            },
+          },
+        },
+        ...commonErrorResponses,
```

Comment:
ftr I don't recommend doing this. It's convenient, but not every api route will surface every one of these errors, it'd be better to explicitely add the ones that can happen.